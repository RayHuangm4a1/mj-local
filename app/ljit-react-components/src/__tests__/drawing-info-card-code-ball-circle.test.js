import React from 'react';
import { shallow, mount, } from 'enzyme';
import CodeBall from '../components/code-ball';
const CodeBallCircle = CodeBall.Circle;
const { SizeEnum, FontSizeEnum, } = CodeBallCircle;

describe('DrawingInfoCardCodeBallCircle', () => {
	let DrawingInfoCardCodeBallCircle;

	beforeEach(() => {
		DrawingInfoCardCodeBallCircle = require('../components/drawing-info-card/elements/drawing-info-card-code-ball-circle').default;
	});

	afterEach(() => {
		DrawingInfoCardCodeBallCircle = undefined;
	});

	it('should handle defaultProps.', () => {
		const {
			size,
			fontSize,
		} = DrawingInfoCardCodeBallCircle.defaultProps;

		expect(size).toEqual(SizeEnum.MIDDLE);
		expect(fontSize).toEqual(FontSizeEnum.MIDDLE);
	});

	it('should be selectable by class drawing-info-card-code-ball-circle', () => {
		const wrapper = shallow(<DrawingInfoCardCodeBallCircle
			splitOpencodes={['1','2','3','4','5',]}
		/>);

		expect(wrapper.hasClass('drawing-info-card-code-ball-circle')).toEqual(true);
	});

	it('should renders correctly', () => {
		const props = {
			splitOpencodes: ['1','2','3','4','5',],
		};

		const wrapper = shallow(<DrawingInfoCardCodeBallCircle {...props} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			splitOpencodes: ['1','2','3','4','5',],
		};

		const wrapper = mount(<DrawingInfoCardCodeBallCircle {...props} />);

		expect(wrapper.props().opencode).toBe(props.opencode);
		expect(wrapper.props().splitOpencode).toBe(props.splitOpencode);
	});
});
