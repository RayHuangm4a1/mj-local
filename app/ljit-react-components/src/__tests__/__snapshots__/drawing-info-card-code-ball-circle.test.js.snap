// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingInfoCardCodeBallCircle should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawingInfoCardCodeBallCircle
    fontSize="middle"
    size="middle"
    splitOpencodes={
      Array [
        "1",
        "2",
        "3",
        "4",
        "5",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Circle
          className="drawing-info-card-code-ball-circle__code-ball"
          fontSize="middle"
          size="middle"
          text="1"
          type="selected"
        />,
        <Circle
          className="drawing-info-card-code-ball-circle__code-ball"
          fontSize="middle"
          size="middle"
          text="2"
          type="selected"
        />,
        <Circle
          className="drawing-info-card-code-ball-circle__code-ball"
          fontSize="middle"
          size="middle"
          text="3"
          type="selected"
        />,
        <Circle
          className="drawing-info-card-code-ball-circle__code-ball"
          fontSize="middle"
          size="middle"
          text="4"
          type="selected"
        />,
        <Circle
          className="drawing-info-card-code-ball-circle__code-ball"
          fontSize="middle"
          size="middle"
          text="5"
          type="selected"
        />,
      ],
      "className": "drawing-info-card-code-ball-circle",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "className": "drawing-info-card-code-ball-circle__code-ball",
          "fontSize": "middle",
          "size": "middle",
          "text": "1",
          "type": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "className": "drawing-info-card-code-ball-circle__code-ball",
          "fontSize": "middle",
          "size": "middle",
          "text": "2",
          "type": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "className": "drawing-info-card-code-ball-circle__code-ball",
          "fontSize": "middle",
          "size": "middle",
          "text": "3",
          "type": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "className": "drawing-info-card-code-ball-circle__code-ball",
          "fontSize": "middle",
          "size": "middle",
          "text": "4",
          "type": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "className": "drawing-info-card-code-ball-circle__code-ball",
          "fontSize": "middle",
          "size": "middle",
          "text": "5",
          "type": "selected",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Circle
            className="drawing-info-card-code-ball-circle__code-ball"
            fontSize="middle"
            size="middle"
            text="1"
            type="selected"
          />,
          <Circle
            className="drawing-info-card-code-ball-circle__code-ball"
            fontSize="middle"
            size="middle"
            text="2"
            type="selected"
          />,
          <Circle
            className="drawing-info-card-code-ball-circle__code-ball"
            fontSize="middle"
            size="middle"
            text="3"
            type="selected"
          />,
          <Circle
            className="drawing-info-card-code-ball-circle__code-ball"
            fontSize="middle"
            size="middle"
            text="4"
            type="selected"
          />,
          <Circle
            className="drawing-info-card-code-ball-circle__code-ball"
            fontSize="middle"
            size="middle"
            text="5"
            type="selected"
          />,
        ],
        "className": "drawing-info-card-code-ball-circle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "className": "drawing-info-card-code-ball-circle__code-ball",
            "fontSize": "middle",
            "size": "middle",
            "text": "1",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "className": "drawing-info-card-code-ball-circle__code-ball",
            "fontSize": "middle",
            "size": "middle",
            "text": "2",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "className": "drawing-info-card-code-ball-circle__code-ball",
            "fontSize": "middle",
            "size": "middle",
            "text": "3",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "className": "drawing-info-card-code-ball-circle__code-ball",
            "fontSize": "middle",
            "size": "middle",
            "text": "4",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "className": "drawing-info-card-code-ball-circle__code-ball",
            "fontSize": "middle",
            "size": "middle",
            "text": "5",
            "type": "selected",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
